libcontainer/rootfs_linux.go

```diff
@@ -152,14 +153,46 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		}
 		return nil
 	case "tmpfs":
+		copyUp := m.Extensions&configs.EXT_COPYUP == configs.EXT_COPYUP
+		tmpDir := ""
 		stat, err := os.Stat(dest)
 		if err != nil {
 			if err := os.MkdirAll(dest, 0755); err != nil {
 				return err
 			}
 		}
-		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
-			return err
+		if copyUp {
+			tmpDirPrefix := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
Want to make this work with read only rootfs that we can't write to at all. 

Sent from my iPhone

> On May 27, 2016, at 3:36 PM, Michael Crosby <notifications@github.com> wrote:
> 
> In libcontainer/rootfs_linux.go:
> 
> >  		stat, err := os.Stat(dest)
> >  		if err != nil {
> >  			if err := os.MkdirAll(dest, 0755); err != nil {
> >  				return err
> >  			}
> >  		}
> > -		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
> > -			return err
> > +		if copyUp {
> > +			tmpDirPrefix := os.Getenv("XDG_RUNTIME_DIR")
> Just use the container's /tmp dir or some hidden dir inside the rootfs to make this. Since its temporary and is only there to make the initial tmpfs destination it can be anywhere or even back in your container's /run.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>